financial-templates-lib/test/price-feed/BalancerPriceFeed.js

```diff
@@ -0,0 +1,45 @@
+const { toWei, toBN } = web3.utils;
+const winston = require("winston");
+
+const { BalancerPriceFeed } = require("../../price-feed/BalancerPriceFeed");
+const { mineTransactionsAtTime, MAX_SAFE_JS_INT } = require("@umaprotocol/common");
+const { delay } = require("../../helpers/delay.js");
+
+const BalancerMock = artifacts.require("BalancerMock");
+const Balancer = artifacts.require("Balancer");
+
+contract("BalancerPriceFeed.js", function(accounts) {
+  const owner = accounts[0];
+
+  let balancerMock;
+  let balancerPriceFeed;
+  let mockTime = 0;
+  let dummyLogger;
+
+  beforeEach(async function() {
+    balancerMock = await BalancerMock.new({ from: owner });
+
+    // DummyLogger will not print anything to console as only capture `info` level events.
+    dummyLogger = winston.createLogger({
+      level: "info",
+      transports: [new winston.transports.Console()]
+    });
+
+    balancerPriceFeed = new BalancerPriceFeed(
+      dummyLogger,
+      web3,
+      () => mockTime,
+      Balancer.abi,
+      balancerMock.address,
+      // These dont matter in the mock, but would represent the tokenIn and tokenOut for calling price feed.
+      accounts[1],
+      accounts[2]
+    );
+  });
+
+  it("Basic current price", async function() {
```

Comment:
It might make sense to test another simple case, or two, just to validate that the update() and getCurrentPrice()/getLastUpdateTime() logic works as the caller expects. Like for instance, it might make sense to test a case where you change the price and verify that you get the new price/time after calling update().

Reply:
Looking through the balance contract, they do have some events. 
- pool swap
- pool join
- pool exit
you would need to have all events up to the block of interest to build the state, but I imagine it might be complicated. What do u think is the best way to do this?